### PR TITLE
feat: the survey api and test

### PR DESCRIPTION
**Pull request by dadiorchen**

### CODE REVIEW

File: api-tests/survey-get-api.spec.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            }]
+          },
+          {
+            labels: ["10 Trees", "11 Trees"],
+            datasets: [{
+              label: "-",
+              data: [1, 1]
+            }]
+          }]);
+      });
```
Comment (dadiorchen) [inline api-tests/survey-get-api.spec.js:96]: The result respects our spec

File: package.json [hunk 2]
```diff
@@ -61,6 +61,7 @@
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-prettier": "^3.2.0",
     "eslint-plugin-simple-import-sort": "^7.0.0",
+    "expect": "^27.5.1",
```
Comment (dadiorchen) [inline package.json:64]: Jest's expect is good in some scenarios.

File: api-tests/survey-get-api.spec.js (hunk 1 above)
Comment (ZavenArra) [reply api-tests/survey-get-api.spec.js:96]: I can just set up a seeding approach in this repo, can you convert your tests to this approach?

File: server/handlers/surveyHandler.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const result = await model.getSurveyReport(messageRepository, req.params.uuid);
+    res.status(200).send(result);
+  } catch (e) {
+    console.log(e);
+    if (session.isTransactionInProgress()) {
+      await session.rollbackTransaction();
+    }
+    res.status(422).json({ ...e });
+  }
+};
```
Comment (ZavenArra) [inline server/handlers/surveyHandler.js:27]: I have standardized the message handler around the concept that repositories are only known to the model, and the session is only know to the service layer.  Could you review how I have the Messages stuff set up and conform this survey report implementation to that pattern?

File: server/models/Message.js [hunk 4]
```diff
@@ -275,8 +275,34 @@ const getMessages = async (session, filterCriteria = undefined) => {
...
+        datasets: [{
+          label: "-",
+          data: [...Object.values(counter)],
+        }]
+      }
+    });
+
+  return surveyReports;
+}
+
```
Comment (ZavenArra) [inline server/models/Message.js:302]: This should go into the Survey model, as it does not require any information about the Message object

File: server/repositories/MessageRepository.js [hunk 5]
```diff
@@ -107,6 +108,24 @@ class MessageRepository extends BaseRepository {
...
+        FROM
+          message
+        WHERE
+          message.survey_id = ?
+          `
+        , [surveyId]
+      );
+    log.warn("result:", result);
+    return result.rows;
+  }
```
Comment (ZavenArra) [inline server/repositories/MessageRepository.js:128]: Please place this in the SurveyRepository

**Review by ZavenArra — CHANGES_REQUESTED** [review]
I cleaned up the architecture of this service, to conform to our goals.  Now that I have merged that code and established the patter, let's make this commit conform to that pattern.

**Comment by dadiorchen** [comment]
@ZavenArra I have done the refactor to follow the new pattern, please take a look.

File: api-tests/survey-get-api.spec.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    //     'title'
+    //   );
+    //       `);
+
+    await request(server)
+      .get(`/survey/${surveySeed.surveyId}`)
+      .set('Accept', 'application/json')
+      .expect(200)
+      .then(res => {
+        log.warn("res.body :", res.body);
```
Comment (ZavenArra) [inline api-tests/survey-get-api.spec.js:87]: No need to use .then() with await.

File: server/handlers/surveyHandler.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  await Joi.object({
+    uuid: Joi.string().required(),
+  }).unknown(false).validateAsync(req.params, { abortEarly: false });
+  const session = new Session();
+  const surveyRepository = new SurveyRepository(session);
+  const surveyQuestionRepository = new SurveyQuestionRepository(session);
+
+  try {
+    const result = await model.getSurveyReport(
+      surveyRepository,
```
Comment (ZavenArra) [inline server/handlers/surveyHandler.js:19]: Still need a refactor here to create a SurveyService to follow the pattern

**Review by ZavenArra — CHANGES_REQUESTED** [review]
A couple changes still needed

**Comment by dadiorchen** [comment]
@ZavenArra made the change.

**Pull request merged**